### DISCUSSION TITLE
ETCD backups with Velero

### DISCUSSION THREAD

**Discussion by andrewkhalaf** [vmware-tanzu/velero#5649] [Community support Q&A]
Is Valero safe to take ETCD backups in a Kubernetes cluster? 
Is the ETCD backup consistent? meaning that if the etcd database is in use (read and write) while taking the velero backup, does Velero take that into account? 
Or does Velero just take a physical backup of the ETCD database files? 

Help is much appreciated.

**Comment by anshulahuja98** [comment]
Hi @andrewkhalaf 
Velero does not take an etcd backup directly
It takes individual dump of each Kubernetes resource by fetching it through the Kube API server and then dumping it.
Hence it does not ensure any consistency of etcd in the backup.

**Reply by andrewkhalaf** [reply]
thank you for explaining. 
If consistency is needed, would you recommend using etcdctl commands to dump the etcd and then using velero's PV restic backup to copy the etcd dump? @anshulahuja98

**Reply by anshulahuja98** [reply]
I have not tried the approach you mentioned, but it seems appropriate as per the scenario you explained.

**Comment by Lyndon-Li** [comment]
@andrewkhalaf 
Could you describe more details of your requirements with etcd backup?

Velero is doing the backup with a finer granularity -- backup individual Kubernetes objects, which basically meets most of the requirements for Kubernetes resource backup.

Therefore, etcd snapshot is not with a high priority to Velero's development, so please just share us your requirements and we will prioritize our work accordingly.
